data/systemd/snapd.apparmor.service.in

```diff
@@ -11,6 +11,7 @@ Before=sysinit.target
 # This dependency is meant to ensure that apparmor initialization (whatever that might entail) is complete.
 After=apparmor.service
 ConditionSecurity=apparmor
+RequiresMountsFor=/var/lib/snapd/apparmor/profiles
```

Comment:
Note you want to do the same for the cache directory. For example, on Ubuntu we use:
```
RequiresMountsFor=/var/cache/apparmor /var/lib/snapd/apparmor/profiles
```

This is actually an interesting point. You could make this distro-agnostic and use `/var/lib/snapd/apparmor/cache` as your cache dir. Then you don't every have to worry about /etc/apparmor.d/cache vs /var/cache/apparmor vs /some/random/distro/dir.

You quite possibly also want:
```
# Don't start this unit on the Ubuntu Live CD
ConditionPathExists=!/rofs/etc/apparmor.d

# Don't start this unit on the Debian Live CD when using overlayfs
ConditionPathExists=!/run/live/overlay/work
```

Reply:
Interesting, I think we could go with both suggestions. I'll send some patches in a moment.

Reply:
Actually, I think the ConditionPathExists comment is wrong. Currently, snapd will in fact load apparmor policy into the kernel when installing snaps so perhaps we skip these conditionals.

Reply:
Nice

Reply:
Done